pkg/sql/sqlinstance/instancestorage/instancestorage.go

```diff
@@ -259,6 +268,64 @@ func (s *Storage) createInstanceRow(
 				return err
 			}
 
+			regionLiveness := regionliveness.NewLivenessProber(s.db, s.codec, nil, s.settings)
+			downRegions, err := regionLiveness.QueryUnavailablePhysicalRegions(ctx, txn, false)
+			if err != nil {
+				return err
+			}
+			if downRegions.ContainsPhysicalRepresentation(string(region)) {
+				// Delete all rows for the region in system.sqlliveness, system.sql_instances,
+				// system.leases.
+				instanceSpan, err := regionliveness.GetSystemTableSpanForRegion(systemschema.SQLInstancesTable(), s.codec, string(region))
+				if err != nil {
+					return err
+				}
+				leaseTableSpan, err := regionliveness.GetSystemTableSpanForRegion(systemschema.LeaseTable(), s.codec, string(region))
+				if err != nil {
+					return err
+				}
+				livenessTableSpan, err := regionliveness.GetSystemTableSpanForRegion(systemschema.SqllivenessTable(), s.codec, string(region))
+				if err != nil {
+					return err
+				}
+				// Save our own session ID before wiping out sessions.
+				sqlLivenessKeyBytesForOurSessionBytes := make([]byte, len(livenessTableSpan.Key))
```

Comment:
Instead of saving the sql server's own session can we run the clean up operation before allocating a session ID?